opentechinstitute/piecewise
# QC check #1 of ? on recent PR series

Since the last round of PRs made significant changes to Piecewise code, I wanted to first test a new deployment to confirm everything is working on Debian 8 hosts. Naming this "QC check # 1 of ?" I also invite others to test deployments and interrogate/improve the codebase.

This PR corrects an issue discovered in a test run, where the **unarchive** module was failing to unzip the MaxMind CSV. It turns out that Maxmind may have introduced a folder structure in their zip archive, which was causing Ansible to fail. Adding the '-j' flag corrects this.

Other additions in this PR include: 
- testing the use of BigQuery project name instead of Project ID # (successfully tested)
- changes the ansible_ssh_user to a non-root account
- a convenience script that creates a GCE VM instance for development: `scripts/setup_gce_vm_instance.sh`

This commit also includes my test VM's IP address and my fork information and has been tested in a GCE VM instance, which you'll need to change to your own VM's IP to test.

To test this code:
- Confirm your gmail account has access to the GCE project mlab-sandbox
- Confirm you have the cloud SDK installed on your machine, and authenticated with your gmail account
- Confirm your cloud SDK has the project `mlab-sandbox` selected
- Run `scripts/setup_gce_vm_instance.sh` and note the VM's external IP
- Update values in these files:
  - ansible/inventory/debian8-hosts : `new-vm-ip-address ansible_ssh_user=your-username-on-your-deployment-machine`
  - ansible/group_vars/all/piecewise_global_config.yml : `site_fqdn: "new-vm-ip-address"`
- If desired, customize additional values in ansible/group_vars/all/piecewise_global_config.yml 
- From the top level folder run: `ansible-playbook -i ansible/inventory/debian8-hosts deploy-piecewise.yml`


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/opentechinstitute/piecewise/124)
